osutil/overlay.go

```diff
@@ -0,0 +1,49 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil
+
+import (
+	"fmt"
+)
+
+// IsRootOverlay returns true if overlayfs is being used for '/'
+//
+// Currently uses variables and Mock functions from nfs.go
+func IsRootOverlay() (bool, error) {
```

Comment:
This is fine. On second look I was thinking that we could return `(bool, upperdir, error)` as we have easy access to this information below.

Reply:
Done, though now I return `(upperdir, error)`.